src/circuit/note_commit.rs

```diff
@@ -2230,129 +2485,196 @@ mod tests {
-                                        )
-                                        .chain(Some(lsb(gd_y_lsb)))
-                                        .chain(
-                                            pkd_x
-                                                .to_le_bits()
-                                                .iter()
-                                                .by_vals()
-                                                .take(L_ORCHARD_BASE),
-                                        )
-                                        .chain(Some(lsb(pkd_y_lsb)))
-                                        .chain(value.to_le_bits().iter().by_vals().take(L_VALUE))
-                                        .chain(
-                                            rho.to_le_bits().iter().by_vals().take(L_ORCHARD_BASE),
-                                        )
-                                        .chain(
-                                            psi.to_le_bits().iter().by_vals().take(L_ORCHARD_BASE),
-                                        ),
-                                    &rcm,
-                                )
-                                .unwrap()
-                                .to_affine()
+                        .zip(self.asset)
+                        .map(|(((g_d, pk_d), (rho, psi)), asset)| {
+                            NoteCommitment::derive(
+                                g_d.to_bytes(),
+                                pk_d.to_bytes(),
+                                value,
+                                asset,
+                                rho,
+                                psi,
+                                NoteCommitTrapdoor(rcm),
+                            )
+                            .unwrap()
+                            .inner()
+                            .to_affine()
                         });
                     NonIdentityPoint::new(ecc_chip, layouter.namespace(|| "witness cm"), point)?
                 };
                 cm.constrain_equal(layouter.namespace(|| "cm == expected cm"), &expected_cm)
             }
         }
 
+        fn affine_point_from_coordinates(x_coord: pallas::Base, y_lsb: pallas::Base) -> EpAffine {
+            // Calculate y = (x^3 + 5).sqrt()
+            let mut y = (x_coord.square() * x_coord + pallas::Affine::b())
+                .sqrt()
+                .unwrap();
+            if bool::from(y.is_odd() ^ y_lsb.is_odd()) {
+                y = -y;
+            }
+            pallas::Affine::from_xy(x_coord, y).unwrap()
+        }
+
         let two_pow_254 = pallas::Base::from_u128(1 << 127).square();
+        let mut rng = OsRng;
+        let random_asset = {
+            let sk = SpendingKey::random(&mut rng);
+            let isk = IssuanceAuthorizingKey::from(&sk);
+            let ik = IssuanceValidatingKey::from(&isk);
+            let asset_descr = "zsa_asset";
+            AssetBase::derive(&ik, asset_descr)
```

Comment:
Repeated here and in `value_commit_orchard.rs`

Reply:
Created AssetBase::random(...)